pkg/reconciler/common/images.go

```diff
@@ -43,7 +43,9 @@ var (
 
 // ImageTransformer is an interface for transforming images passed to the ResourceImageTransformer
 type ImageTransformer interface {
-	HandleImage(containerName, parent string, useDefault bool) (string, bool)
+	ImageForContainer(container *corev1.Container, parentName string) (string, bool)
```

Comment:
Should/could we have a default impl that only requires you to populate this method (for callers who don't have different behavior between env vars, cached images, and containers)?

Reply:
Spoke offline, the main issue is the use of env var name as container names (which may be a bug?)

Can you add a TODO to investigate that?

Reply:
Fixed. Changed to resource name